backend_plugin/src/main/evaluators/BlockCommentEvaluator.java

```diff
@@ -29,72 +34,52 @@ public BlockCommentEvaluator() {
-
-				// Verify that the double backslash is at the start of the line
-				IDocument doc = event.getDocument();
-				try {
-					int lineOffset = doc.getLineInformationOfOffset(currentOffset).getOffset();
-					String textBeforeSlashes = doc.get(lineOffset, prevOffset-lineOffset);
-
-					// DEBUG
-					//					System.out.println("line offset: " + lineOffset);
-					//					System.out.println("Prev offset: " + prevOffset);
-					//					System.out.println("Text before slashes: " + textBeforeSlashes);
+		if (doubleBackSlash(event)) {
+			IDocument doc = event.getDocument();
+			try {
+				// Verify that the double slash is at the beginning of the line
+				int lineOffset = doc.getLineInformationOfOffset(currentOffset).getOffset();
+				String textBeforeSlashes = doc.get(lineOffset, prevOffset-lineOffset);
 
-					textBeforeSlashes = textBeforeSlashes.trim();
-					if (textBeforeSlashes.isEmpty()) {
+				textBeforeSlashes = textBeforeSlashes.trim();
+				if (textBeforeSlashes.isEmpty()) {
 
-						// DEBUG
-//						System.out.println("Start of line!");
+					// If we have two back slashes, then check if we've found two previously
+					if (firstBackSlashDetected) {
+						IRegion currentRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
 
-						// If we have two back slashes, then check if we've found two previously
-						if (firstBackSlashDetected) {
-							IRegion currentRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
+						// Check if consecutive lines
+						int prevToCurrentDif = currentRegion.getOffset() - (prevRegion.getOffset() + prevRegion.getLength());
+						int currentToPrevDif = prevRegion.getOffset() - (currentRegion.getOffset() + currentRegion.getLength());
 
-							// DEBUG
-							// System.out.println("Current Region pos: " + currentRegion.getOffset());
-							// System.out.println("Current REgion length: " + currentRegion.getLength());
-							// System.out.println("Prev Region pos: " + prevRegion.getOffset());
-							// System.out.println("Prev Region length: " + prevRegion.getLength());
-
-							// Check if consecutive lines
-							int prevToCurrentDif = currentRegion.getOffset() - (prevRegion.getOffset() + prevRegion.getLength());
-							int currentToPrevDif = prevRegion.getOffset() - (currentRegion.getOffset() + currentRegion.getLength());
-
-							// Uses a buffer of 2 characters because sometimes IRegion doesn't count the newly inserted "//"
-							if ((prevToCurrentDif >= 0 && prevToCurrentDif <= 2) ||
-									(currentToPrevDif >= 0 && currentToPrevDif <= 2)) {
-								firstBackSlashDetected = false;
-								return true;
-							}
-
-							// If not consecutive, then update prevRegion
+						// Uses a buffer of 2 characters because sometimes IRegion doesn't count the newly inserted "//"
+						if ((prevToCurrentDif >= 0 && prevToCurrentDif <= 2) ||
+								(currentToPrevDif >= 0 && currentToPrevDif <= 2)) {
 							prevRegion = currentRegion;
```

Comment:
We assign `prevRegion = currentRegion` outside of this if-else block now, so we only need the else case